src/engine/engine.h

```diff
@@ -6,4 +6,28 @@
 namespace mixxx {
     // TODO(XXX): When we move from stereo to multi-channel this needs updating.
     static constexpr mixxx::AudioSignal::ChannelCount kEngineChannelCount(2);
+
+    // Contains the information needed to process a buffer of audio
+    class AudioParameters : public AudioSignal {
+      public:
+        SINT framesPerBuffer() const {
+            return m_framesPerBuffer;
+        }
+        SINT bufferSize() const {
+            return static_cast<SINT>(channelCount()) * framesPerBuffer();
+        }
+
+        explicit AudioParameters(
+                SampleLayout sampleLayout,
+                ChannelCount channelCount,
```

Comment:
Do we actually use this parameter in the constructor or shouldn't this be hard-coded to kEngineChannelCount until we actually need it? The same is true for sampleLayout which currently is always interleaved. Currently this and the maximum sample rate constant is hidden somewhere in effectchain.cpp

This constructor does not need to be explicit if it has more than a single parameter.

Reply:
Sure, that makes sense to move that assumption to the central engine.h file.